components/brave_rewards/resources/page/storage.ts

```diff
@@ -12,7 +12,7 @@ export const defaultState: Rewards.State = {
   enabledAds: false,
   enabledAdsMigrated: false,
   enabledContribute: false,
-  firstLoad: null,
+  firstLoad: false,
```

Comment:
Is this causing the first bug you mentioned?
> The current inline tipping settings are never fetched from the rewards service.

Reply:
Basically just trying to understand how this would affect
https://github.com/brave/brave-core/blob/ef8a0cd83aa68116654f3a8eb65fb3feabfedd1b/components/brave_rewards/resources/page/components/settingsPage.tsx#L71-L81

Reply:
The usage of `firstLoad` appears to be a left-over from when Rewards still had a master toggle, and has not worked correctly since. It should probably be removed altogether; I'll see if we can make that work.

For testing, we'd have to mock out the `chrome.send(...` API and responses, which we aren't doing at the moment and which might take some time. Eventually the plan is to update the browser-to-JS layer so that it is more easily testable.

Reply:
ah ok - no worries then 😄 Thanks for providing more info about the context